tests/run_all_tests.py

```diff
@@ -1,63 +1,511 @@
+            if self.action == 'run_tests':
+                self.global_res += test_obj.run_all_test_flow(self.all_results)
+            elif self.action == 'override_expected_output':
+                test_obj.update_expected_output()
+
+    # given a scheme file or a cmdline file, run all relevant tests
+    def run_test_per_file(self, test_file):
+        if self.test_files_spec.type == 'scheme':
+            if self.tests_type in {'general', 'fw_rules_assertions'}:
+                scheme_obj_list = [SchemeFile(test_file, self._get_scheme_test_args(test_file))]
+            else:
+                scheme_obj_list = self.get_scheme_obj_list_for_test(test_file)
+            self.create_and_run_test_obj(scheme_obj_list, 0)
+
+        elif self.test_files_spec.type == 'cmdline':
+            with open(test_file) as doc:
+                code = YAML().load_all(doc)
+                for test in next(iter(code)):
+                    query_name = test.get('name', '')
+                    cli_test_name = f'{os.path.basename(test_file)}, query name: {query_name}'
+                    cliQuery = CliQuery(test, self.test_files_spec.root, cli_test_name)
+                    self.create_and_run_test_obj([cliQuery], test.get('expected', None))
+
+    @staticmethod
+    def _get_scheme_test_args(test_file, out_format_arg=None, out_path_arg=None):
+        res = ['--scheme', test_file]
+        if out_format_arg is not None:
+            res.append('-o')
+            res.append(out_format_arg)
+        if out_path_arg is not None:
+            res.append('-f')
+            res.append(out_path_arg)
+        return res
+
+    # may require to create multiple tests for scheme file, if out_format_arg_list has > 1 out formats to test with
+    def get_scheme_obj_list_for_test(self, test_file):
+        res = []
+        if self.test_files_spec.out_format_arg:
+            for out_format in self.test_files_spec.out_format_arg:
+                out_file_arg = self.get_out_file_arg(test_file, out_format)
+                test_args = self._get_scheme_test_args(test_file, out_format, out_file_arg)
+                scheme_obj = SchemeFile(test_file, test_args)
+                res.append(scheme_obj)
+        else:
+            out_file_arg = self.get_out_file_arg(test_file)
+            test_args = self._get_scheme_test_args(test_file, None, out_file_arg)
+            scheme_obj = SchemeFile(test_file, test_args)
+            res = [scheme_obj]
+        return res
+
+    # relevant for a scheme file, when overriding outputConfig with out_file_arg from cli
+    def get_out_file_arg(self, test_file, out_format='txt'):
+        if self.test_files_spec.add_out_path_arg:
+            suffix = f'{self.test_files_spec.out_path_arg_suffix}.{out_format}'
+            return os.path.basename(test_file).replace(".yaml", suffix)
+        return None
+
+    @staticmethod
+    def get_actual_output_dir():
+        return os.getcwd()
```

Comment:
We should probably have a dedicated directory for actual outputs (in our git repository, this directory should contain only a `.gitignore` file which ignores all files). This way, local repository can be kept clean.

Reply:
done